src/main/java/com/rapid7/client/dcerpc/service/dto/SID.java

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright 2017, Rapid7, Inc.
+ *
+ * License: BSD-3-clause
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * * Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright
+ * notice, this list of conditions and the following disclaimer in the
+ * documentation and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ */
+package com.rapid7.client.dcerpc.service.dto;
+
+import java.util.Arrays;
+import org.bouncycastle.util.encoders.Hex;
+
+/**
+ * Security Identifier (SID) as defined in https://msdn.microsoft.com/en-us/library/cc246018.aspx
+ */
+public class SID {
+    final private byte revision;
+    final private char subAuthorityCount;
+    final private byte[] identifierAuthority;
+    final private long[] subAuthority;
+
+    public SID(byte revision, byte[] identifierAuthority, long[] subAuthority) {
+        this.revision = revision;
+        this.identifierAuthority = identifierAuthority;
+        this.subAuthorityCount = (char) subAuthority.length;
+        this.subAuthority = subAuthority;
+    }
+
+    public SID addRelativeId(long rid){
+        if (rid != 0) {
```

Comment:
I don't think the DTO should make the judgement whether it's valid or not. We should take whatever long that's given. If it's broken it can break when it does. It's not the job of this DTO to verify. It's also confusing when you are returning itself after a successful addRelativeID call, the user doesn't know nothing happened and he's still getting the exact same SID.

Reply:
Fair enough. I'll change it so we can accept all RIDs